x-pack/legacy/plugins/siem/public/containers/detection_engine/signals/types.ts

```diff
@@ -38,3 +42,68 @@ export interface UpdateSignalStatusProps {
   kbnVersion: string;
   signal?: AbortSignal; // TODO: implement cancelling
 }
+
+// export class PostSignalsError extends Error {
+//   constructor(m: string) {
+//     super(m);
+//     // Set the prototype explicitly.
+//     Object.setPrototypeOf(this, SignalsIndexError.prototype);
+//   }
+// }
+
+export interface SignalsIndex {
+  name: string;
+}
+
+export interface Privilege {
+  username: string;
+  has_all_requested: boolean;
+  cluster: {
+    monitor_ml: boolean;
+    manage_ccr: boolean;
+    manage_index_templates: boolean;
+    monitor_watcher: boolean;
+    monitor_transform: boolean;
+    read_ilm: boolean;
+    manage_api_key: boolean;
+    manage_security: boolean;
+    manage_own_api_key: boolean;
+    manage_saml: boolean;
+    all: boolean;
+    manage_ilm: boolean;
+    manage_ingest_pipelines: boolean;
+    read_ccr: boolean;
+    manage_rollup: boolean;
+    monitor: boolean;
+    manage_watcher: boolean;
+    manage: boolean;
+    manage_transform: boolean;
+    manage_token: boolean;
+    manage_ml: boolean;
+    manage_pipeline: boolean;
+    monitor_rollup: boolean;
+    transport_client: boolean;
+    create_snapshot: boolean;
+  };
+  index: {
+    [indexName: string]: {
+      all: boolean;
+      manage_ilm: boolean;
+      read: boolean;
+      create_index: boolean;
+      read_cross_cluster: boolean;
+      index: boolean;
+      monitor: boolean;
+      delete: boolean;
+      manage: boolean;
+      delete_index: boolean;
+      create_doc: boolean;
+      view_index_metadata: boolean;
+      create: boolean;
+      manage_follow_index: boolean;
+      manage_leader_index: boolean;
+      write: boolean;
+    };
+  };
+  isAuthenticated: true;
```

Comment:
Did you mean `boolean` here?

```suggestion
  isAuthenticated: boolean;
```